src/main/java/pl/put/poznan/json/tools/model/decorations/NullRemover.java

```diff
@@ -4,14 +4,46 @@
 import pl.put.poznan.json.tools.model.JsonObject;
 import pl.put.poznan.json.tools.service.WrongInputException;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class NullRemover extends JsonDecorator {
 
     public NullRemover(JsonObject jsonObject) {
         super(jsonObject);
     }
 
     public String getJson() throws WrongInputException {
-        // jsonObject.getJson() zwraca String który musisz przetworzyć odpowiednio i zwrócić
-        return jsonObject.getJson();
+        String jsonString                               = jsonObject.getJson();
+        jsonString                                      = jsonString.substring(1, jsonString.length() - 1);
+        List<String> jsonAttributes                     = Arrays.asList(jsonString.split("\\s*,\\s*"));
+        ArrayList<List<String>> jsonAttributesSplited   = new ArrayList<>();
+        ArrayList<String> jsonAttributesRemovedNulls    = new ArrayList<>();
+
+        splitAttributes(jsonAttributes, jsonAttributesSplited);
+        removeNulls(jsonAttributesSplited, jsonAttributesRemovedNulls);
+
+        String jsonFinal = "{\n\t" + String.join(",\n\t", jsonAttributesRemovedNulls) + "\n}";
+
+        return jsonFinal;
+    }
+
+    private void splitAttributes(List<String> originalList, ArrayList<List<String>> splitedList) {
+        originalList.forEach(item -> splitedList.add(Arrays.asList(item.split("\\s*: \\s*"))));
+
+        splitedList.forEach(item -> {
+            item.set(0, item.get(0).substring(item.get(0).indexOf("\""), item.get(0).length()));
+
+            if(item.get(1).lastIndexOf("\"") != -1) item.set(1, item.get(1).substring(0, item.get(1).lastIndexOf("\"") + 1));
+            else item.set(1, item.get(1).replaceAll("\\s+", ""));
+        });
+    }
+
+    private void removeNulls(ArrayList<List<String>> originalList, ArrayList<String> removedNullsList) {
```

Comment:
ta sama uwaga co do argumentów poprzedniej metody - to zdecydowanie nie powinien być void